### PR TITLE
20 enable calling more than one service

### PR DESCRIPTION
**Pull request by pawelkoper**
---

I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,9 +64,9 @@ The following table describes all elements and attributes used in the template.
 | Element                             | Description                                                              |
 | ----------------------------------- | ------------------------------------------------------------------------ |
 | `data-api-type="templating"`        | required for **Knot.x** to recognize the script as a template to process |
-| `data-call-uri`                     | path to a microsevice that provides the data - it will be handled by a service, as described in the [Configuration](#configuration) section |
+| `data-call-uri.dataservice`         | path to a microsevice that provides the data - it will be handled by a service, as described in the [Configuration](#configuration) section. Last attribute part ('dataservice') is an optional namespace. Only placeholders with matching namespace will be filled by data coming from that service|
```
Comment (Skejven) [inline README.md]: Should probably be `data-call-uri-dataservice`?

File: knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceEntry.java [hunk 2]
```diff
@@ -20,12 +20,18 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.jsoup.nodes.Attribute;
 
+import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
 
 public class ServiceEntry {
+    private static final String NAMESPACE_SEPARATOR = "_";
```
Comment (Skejven) [inline knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceEntry.java]: Is this constant used anywhere?

File: knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceAttributeUtil.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+final class ServiceAttributeUtil {
+
+    private static final int NAMESPACE_GROUP_INDEX = 4;
+
+    private static final String ATTR_PATTERN = "data-call-uri(-(post|get|all))?(-(\\w*))?";
```
Comment (Skejven) [inline knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceAttributeUtil.java]: `(post|get|all)` - this is for the future implementation right?

File: README.md (hunk 1 above)
Comment (pawelkoper) [inline README.md]: fixed

File: knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceAttributeUtil.java (hunk 3 above)
Comment (pawelkoper) [inline knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceAttributeUtil.java]: yes. It is used in https://github.com/Cognifide/knotx/pull/37

File: knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceEntry.java (hunk 2 above)
Comment (pawelkoper) [inline knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceEntry.java]: no. removed

**Comment by Skejven** [comment]
Approved

**Pull request merged**